### ISSUE TITLE
Potential use-after-move in raft_group_registry (courtesy of clang-tidy)

### ISSUE THREAD

**Issue by raphaelsc** [scylladb/scylladb#18331]
```
49 warnings generated.
service/raft/raft_group_registry.cc:543:66: warning: 'id' used after it was moved [bugprone-use-after-move]
            auto& rate_limit = _rate_limits.try_get_recent_entry(id, std::chrono::minutes(5));
                                                                 ^
service/raft/raft_group_registry.cc:539:19: note: move occurred here
    auto dst_id = raft::server_id{std::move(id)};
                  ^
```

I don't think it's a problem today, but it's fragile to rely on a certain behavior of direct_failure_detector::pinger::endpoint_id's move ctor.

**Comment by raphaelsc** [comment]
@tchaikov you fixed this one right? if so, please close this.

**Comment by tchaikov** [comment]
fixed by 8168f02550ada9c9da7e54ac21b353e9fa557290. closing.
